libparsec/crates/client/src/certif/user_update_profile.rs

```diff
@@ -0,0 +1,179 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_client_connection::ConnectionError;
+use libparsec_protocol::authenticated_cmds;
+use libparsec_types::prelude::*;
+
+use super::{
+    greater_timestamp, store::CertifStoreError, CertificateBasedActionOutcome, CertificateOps,
+    GreaterTimestampOffset, InvalidCertificateError, InvalidKeysBundleError,
+};
+use crate::EventTooMuchDriftWithServerClock;
+
+#[derive(Debug, thiserror::Error)]
+pub enum CertifUpdateUserProfileError {
+    #[error("Component has stopped")]
+    Stopped,
+    #[error("Cannot reach the server")]
+    Offline,
+    #[error("Cannot revoke oneself")]
+    UserIsSelf,
+    #[error("User not found")]
+    UserNotFound,
+    #[error("Author not allowed")]
+    AuthorNotAllowed,
+    #[error("User revoked")]
+    UserRevoked,
+    #[error("Our clock ({client_timestamp}) and the server's one ({server_timestamp}) are too far apart")]
+    TimestampOutOfBallpark {
+        server_timestamp: DateTime,
+        client_timestamp: DateTime,
+        ballpark_client_early_offset: f64,
+        ballpark_client_late_offset: f64,
+    },
+    #[error("There is no key available in this realm for encryption")]
+    NoKey,
```

Comment:
Please, take some extra time when creating an error to make sure the variants are meaningful for the use case :pray: 

The whole point of having each function with a dedicated error is to precisely describe what can go wrong. I strongly suggest you to any new error empty, then gradually add items to it instead of copy/pasting from another one.